File: frontoffice/app/src/main/java/com/ds201625/fonda/domains/DishOrder.java
package com.ds201625.fonda.domains;

/**
 * Created by Katherina Molina on 10/05/2016.
 */

/*
 Clase quee representa el pedido del cliente
 */
public class DishOrder extends BaseEntity {

    /**
     * Dish que define el dish asociado a la cuenta
     */
    private Dish dish;

    /**
     * Integer que define la cantidad de platos
     */
    private int count;

    /**
     Contructor de la clase PlatoOrdenado
     */
    public DishOrder() {
        super();
    }

    public DishOrder(Dish dish, int count) {
        this.dish = dish;
        this.count = count;
    }


    /**
     * Metodo que obtiene el plato de la orden
     * @return dish
     */
    public Dish getDish() {
        return dish;
    }

    /**
     * Metodo que asigna el plato a la orden
     * @param dish
     */
    public void setDish(Dish dish) {
        this.dish = dish;
    }

    /**
     * Metodo que obtiene la cantidad del plato de la orden
     * @return count
     */
    public int getCount() {
        return count;
    }

    /**
     * Metodo que asigna la cantidad del plato a la orden
     * @param count
     */
    public void setCount(int count) {
        this.count = count;
    }
}

File: backoffice/FondaDomain/Domain/DishOrder.cs
using System;
using System.Runtime.Serialization;

namespace com.ds201625.fonda.Domain
{
    /// <summary>
    /// Representa la orden de una cuenta
    /// </summary>
    public class DishOrder : BaseEntity
    {

        #region Fields

        /// <summary>
        /// Plato ordenado
        /// </summary>
        private Dish _dish;

        /// <summary>
        /// Cantidad de veces que se ordeno el plato
        /// </summary>
        private int _count;

        #endregion

        #region Constructors

        /// <summary>
        /// Constructor vacio
        /// </summary>
        public DishOrder() : base () { }

        #endregion

        #region Properties

        /// <summary>
        /// Obtiene o asigna la cantidad de un plato
        /// </summary>
        [DataMember]
        public virtual int Count
        {
            get { return _count;  }
            set { _count = value; }
        }

        /// <summary>
        /// Obtiene o asigna un plato
        /// </summary>
        [DataMember]
        public virtual Dish Dish
        {
            get { return _dish; }
            set { _dish = value; }
        }

        /// <summary>
        /// Obtiene el precio del plato
        /// </summary>
        public virtual float Dishcost
        {
            get { return _dish.Cost; }
        }

        #endregion 
    }
}
